### DISCUSSION TITLE
USDRT in IsaacLab / interacting with Fabric

### DISCUSSION THREAD

**Discussion by AntoineRichard** [isaac-sim/IsaacLab#1304] [Q&A]
Hi there,

I'm trying to use USDRT to talk to Fabric in IsaacLab, but it does not seem to work. It's as if the Fabric stage I'm getting is unable to see the regular USD stage, or even the fabric stage used by the rest of the framework. Is this expected?

What's the proper way to interact with Fabric? Do I need to use the C++ API?

Let me know,

Best,

Antoine

**Comment by mmonem1996** [comment]
Hello Antoine,

I am facing the same issue. I am using usdrt C++ API to talk to fabric (query information about transforms using usdrt::UsdStage). 

It it is also not working. It is linked to "omni.physx.fabric" extension.. The API I am using are able to access transient simulation results if the extension is disabled (in this case, the USD stage is updated during the simulation, which defeats the purpose). For some reason, I can only disable/enable this extention in IsaacSim but not through IsaacLab.

I tried using Fabric's API (e.g. StageReaderWriter) hoping I get different results but faced linking errors.

Best
Mohammed

**Comment by AntoineRichard** [comment]
Hey! Good to know, I was tempted to see if it was a python USDRT issue and go the C++ route, but I've been too lazy to do it yet. 
USDRT worked flawlessly in python in IsaacSim, but it doesn't in IsaacLab. If you do get it to work i'm interested, in particular if you get it to interact with physics enabled objects after the initial setup. 

Best,

Antoine

**Reply by mmonem1996** [reply]
Hello! Ironically I was also thinking I should try Python as well. Actually it is not fully in C++. I was not able to get the usd stage id in C++ without running into compilation issues. So I made a small script to send the ID to be able to test the code I wrote(lol) .
I will update here once I figure out something, and I am also interested to see what you get

Many thanks,

Mohammed

**Reply by AntoineRichard** [reply]
Oh, I see. My main target is building static colliders that can be moved on an environment reset. I was hoping a PointInstancer could be used (in USDRT/Fabric) and it works in IsaacSim, but not in IsaacLab. I'll let you know if I manage.

Cheers,

Antoine
